src/java/org/apache/cassandra/config/Config.java

```diff
@@ -257,11 +257,14 @@
     public volatile long native_transport_max_concurrent_connections_per_ip = -1L;
     public boolean native_transport_flush_in_batches_legacy = false;
     public volatile boolean native_transport_allow_older_protocols = true;
-    public volatile long native_transport_max_concurrent_requests_in_bytes_per_ip = -1L;
-    public volatile long native_transport_max_concurrent_requests_in_bytes = -1L;
+    @Replaces(oldName = "native_transport_max_concurrent_requests_in_bytes_per_ip", converter = Converters.BYTES_CUSTOM_DATASTORAGE, deprecated = true)
+    public volatile DataStorageSpec native_transport_max_concurrent_requests_per_ip = new DataStorageSpec("0B");
+    @Replaces(oldName = "native_transport_max_concurrent_requests_in_bytes", converter = Converters.BYTES_CUSTOM_DATASTORAGE, deprecated = true)
+    public volatile DataStorageSpec native_transport_max_concurrent_requests = new DataStorageSpec("0B");
     public volatile boolean native_transport_rate_limiting_enabled = false;
     public volatile int native_transport_max_requests_per_second = 1000000;
     // not exposed in the yaml
+    // not sure about this one.... -1? Negative? I don't see anything that guards it?
     public int native_transport_receive_queue_capacity_in_bytes = 1 << 20; // 1MiB
```

Comment:
@dcapwell any thoughts here?

Reply:
Used by `org.apache.cassandra.net.AbstractMessageHandler`

```
if (currentQueueSize + bytes <= queueCapacity)
        {
            queueSizeUpdater.addAndGet(this, bytes);
            return ResourceLimits.Outcome.SUCCESS;
        }
```

```
long allocatedExcess = min(currentQueueSize + bytes - queueCapacity, bytes);
```

```
long actualExcess = max(0, min(newQueueSize - queueCapacity, bytes));
```

```
if (oldQueueSize > queueCapacity)
        {
            long excess = min(oldQueueSize - queueCapacity, bytes);

            endpointReserveCapacity.release(excess);
            globalReserveCapacity.release(excess);

            endpointWaitQueue.signal();
            globalWaitQueue.signal();
        }
```

As far as I can tell tell (review, not tested), -1 will have weird and unexpected behavior for that class... so I see it as unexpected and error prone...

Blocking < 0 makes sense to me, let me also ask Sam (author)

Reply:
Sam said it is cool to update, and that a subclass already was updated via `public DataStorageSpec internode_application_receive_queue_capacity = new DataStorageSpec("4MiB");`

So `public DataStorageSpec native_transport_receive_queue_capacity = DataStorageSpec.inMebibytes(1);` would be safe